--- a/obmenu-generator-0.89/obmenu-generator	2021-08-22 20:13:48.916371343 +0200
+++ b/obmenu-generator-0.89/obmenu-generator	2021-08-22 20:04:45.799719966 +0200
@@ +1,6 -1,6 @@
 #!/usr/bin/perl
 
-# Copyright (C) 2010-2020 Daniel "Trizen" Șuteu <echo dHJpemVuQHByb3Rvbm1haWwuY29tCg== | base64 -d>.
+# Copyright (C) 2010-2021 Daniel "Trizen" Șuteu <echo dHJpemVuQHByb3Rvbm1haWwuY29tCg== | base64 -d>.
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -24,7 +24,7 @@
 # Name: obmenu-generator
 # License: GPLv3
 # Created: 25 March 2011
-# Latest edit: 03 December 2020
+# Latest edit: 10 August 2021
 # https://github.com/trizen/obmenu-generator
 
 use 5.014;
@@ -450,7 +450,7 @@
         $cache_db{__GTKRC_MTIME__} = $gtkrc_mtime;
     }
     else {
-        my $old_mtime = $cache_db{__GTKRC_MTIME__} // -1;
+        my $old_mtime = exists($cache_db{__GTKRC_MTIME__}) ? $cache_db{__GTKRC_MTIME__} : -1;
         if ($old_mtime != $gtkrc_mtime) {
             print STDERR ":: Regenerating the cache DB...\n";
 
@@ -469,7 +469,11 @@
     my %fast_cache;
 
     sub check_icon {
-        $fast_cache{$_[0] // return undef} //= ($cache_db{$_[0]} //= get_icon_path($_[0]));
+        $fast_cache{$_[0] // return undef} //= do {
+            exists($cache_db{$_[0]})
+              ? $cache_db{$_[0]}
+              : do { $cache_db{$_[0]} = get_icon_path($_[0]) }
+        };
     }
 }
 
@@ -489,7 +493,7 @@
 my %categories;
 foreach my $file ($desk_obj->get_desktop_files) {
 
-    my %info = split("\0\1\0", $cache_db{$file} // '', -1);
+    my %info = split("\0\1\0", (exists($cache_db{$file}) ? $cache_db{$file} : ''), -1);
 
     next if exists $info{__IGNORE__};
 
